### PR TITLE
Object Comparison methods

### PR DESCRIPTION
**Pull request by mjfox3**
I added __eq__ methods to the tiers and textgrid object in order to facilitate unit test construction for projects that incorporate textgrids. For example, projects that have read and write methods for textgrids for both the native .TextGrid format and for writing as .csv, can now test that reading in one format and writing in another produce the expected results. Unit tests are included.

### CODE REVIEW

File: textgrid/textgrid.py [hunk 1]
```diff
@@ -314,6 +314,14 @@ def __init__(self, name=None, minTime=0., maxTime=None):
         self.maxTime = maxTime
         self.points = []
 
+    def __eq__(self, other):
+        if not hasattr(other, 'points'):
+            return False
+        elif all(a == b for a, b in zip(other.points, self.points)):
```
Comment (kylebgorman) [inline textgrid/textgrid.py]: I would do:

else:
    return all(a == b for a, b in zip(self.points, other.points))

That eliminates the true/false switch and I think it's natural to have `self` before `other`.

File: textgrid/textgrid.py [hunk 2]
```diff
@@ -297,7 +297,7 @@ def __contains__(self, other):
             return self.minTime <= other <= self.maxTime
 
     def bounds(self):
-        return (self.minTime, self.maxTime)
+        return self.minTime, self.maxTime
```
Comment (kylebgorman) [inline textgrid/textgrid.py]: Can you easily un-revert the deleted parens? I love parens around tuples.

File: textgrid/textgrid.py [hunk 3]
```diff
@@ -422,6 +430,14 @@ def __init__(self, name=None, minTime=0., maxTime=None):
         self.intervals = []
         self.strict = True
 
+    def __eq__(self, other):
+        if not hasattr(other, 'intervals'):
+            return False
+        elif all(a == b for a, b in zip(other.intervals, self.intervals)):
```
Comment (kylebgorman) [inline textgrid/textgrid.py]: ditto from above

File: textgrid/textgrid.py [hunk 4]
```diff
@@ -606,6 +622,14 @@ def __init__(self, name=None, minTime=0., maxTime=None, strict=True):
         self.tiers = []
         self.strict = strict
 
+    def __eq__(self, other):
```
Comment (kylebgorman) [inline textgrid/textgrid.py]: ditto from above

File: textgrid/textgrid.py (hunk 1 above)
Comment (mjfox3) [reply textgrid/textgrid.py]: Done.

File: textgrid/textgrid.py (hunk 2 above)
Comment (mjfox3) [reply textgrid/textgrid.py]: I reverted the removal.

File: textgrid/textgrid.py (hunk 3 above)
Comment (mjfox3) [reply textgrid/textgrid.py]: Done.

**Comment by mjfox3** [comment]
I've added another commit to address your change requests.

**Comment by kylebgorman** [comment]
Merging now.

**Pull request merged**